pandasql/core.py

```diff
@@ -47,7 +50,13 @@ def __init__(self, name=None, sources=None):
     @property
     def result(self):
         if self._cached_result is None:
-            return None
+            if self._out_of_memory:
+                raise MemoryError('The result of this dataframe is too big to '
+                                  'fit in memory. Please try accessing a '
+                                  'smaller subset of the data you need, '
+                                  'e.g., using df.head().')
```

Comment:
Might be a better UX if we automatically give df.head() and show a warning?

Reply:
I worry that it'll be unintuitive that the code "worked" but had only the top 10 rows. Not a fan of soft failures for a scenario like this.